lib/_stream_writable.js

```diff
@@ -378,7 +378,7 @@ function writeOrBuffer(stream, state, isBuf, chunk, encoding, cb) {
   if (!ret)
     state.needDrain = true;
 
-  if (state.writing || state.corked) {
+  if (state.writing || state.corked || state.bufferedRequestCount > 0) {
```

Comment:
This is definitely wrong. It will make it literally impossible to ever buffer requests because the if body is the only place where this value gets incremented. More to the point, it would break the basic promise of streams that there's only ever one actual write on the wire.